components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1223,4 +1235,200 @@ private static boolean isSelfSignedCert(X509Certificate cert) {
         }
     }
 
+    private static void addDefaultValidatorConfigToConfigStore(OMElement validatorsElement, String tenantDomain)
+            throws CertificateValidationException {
+        List<Validator> defaultValidatorConfig = getDefaultValidatorConfig(validatorsElement);
+
+        for (Validator validator : defaultValidatorConfig) {
+            String validatorResourceType = X509CertificateValidationConstants.VALIDATOR_RESOURCE_TYPE;
+            if (log.isDebugEnabled()) {
+                log.debug("Adding default validator configurations to config store in: " + validatorResourceType);
+            }
+            try {
+                CertValidationDataHolder.getInstance().getConfigurationManager()
+                        .getResourcesByType(validatorResourceType);
+            } catch (ConfigurationManagementException e) {
+                addResourceTypeIfNotExists(e, validatorResourceType, validator.getDisplayName(), tenantDomain);
+            }
```

Comment:
This part adds the `resourceType()` and we only needs to do it once.. So shouldnt we have this outside of the for loop?

Reply:
removed getResourceType since we are adding the types through db scripts